src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -355,6 +371,8 @@ def _execute_live_or_recording(self):
             if callable(tear_down) and not self.skip_teardown:
                 self.tear_down()
 
+    @mock.patch('azure.cli.command_modules.datalake.analytics.custom._get_uuid_str',
+                _mock_get_uuid_str)
```

Comment:
That makes sense, but the problem here is that the UUID is generated in the function submit_adla_job() which is part of the custom.py in the product code. My test isn't aware of what the job ID is until after it calls the submit command, and the patch is what ensures that the UUID used is the "right" one. Where can I put the line above so that it behaves the same way?